### PR TITLE
Adds Exclusive Tag to dynamic rulesets & Makes Blob an exclusive ruleset

### PR DESCRIPTION
**Pull request by DorainGrey**
[gamemode] [wip]

Very experimental, not entirely sure how to test this or whether this is the best way to do it.

## What this does
Adds Exclusive events to the dynamic ruleset datums, events that are exclusive will be drafted alone and no other rulesets will be injected on roundstart unless the server population is greater than 45. Important to note, this does not affect midround injections or midround blob.

## Why it's good
Blob is primarily a team gamemode, it's also designed to be a standalone roundender. 
Spawning traitors often ruins the gamemode for both the blob and the normal station crew.
 
## Changelog
:cl:
 * rscadd: Added new thing.
 * tweak: Changed a 5 to a 6.

### CODE REVIEW

**Comment by Hinaichigo** [comment]
Blobs should be high enough threat that getting a blob with traitors would be relatively rare, but not explicitly prohibited. Fewer possible ruleset combinations, and less potential for chaos, goes against the idea of dynamic I feel.
Also it would facilitate metagaming; if you know there's a blob, you know there's no traitors, and vice versa (for roundstart anyway).

**Comment by Kurfursten** [comment]
Two of my most memorable rounds ever were blob+tot. Worse, this means if blob fires you know there cannot possibly be any other antags, even after the blob is dead. I have found historically the interplay of blob and traitor is an interesting one so I'm not sure why you say it ruins things for both sides. Complicates things? Sure. Arming everyone when someone might not have good intentions IS a complication you should keep in mind. That's why liberators were made. Too bulky to conceal, reduced damage to humans.

Also, the CL is terrible.

**Comment by DorainGrey** [comment]
> Two of my most memorable rounds ever were blob+tot. Worse, this means if blob fires you know there cannot possibly be any other antags, even after the blob is dead. I have found historically the interplay of blob and traitor is an interesting one so I'm not sure why you say it ruins things for both sides. Complicates things? Sure. Arming everyone when someone might not have good intentions IS a complication you should keep in mind. That's why liberators were made. Too bulky to conceal, reduced damage to humans.

This should allow midround injections after, it does not eat the threat nor prevent midrounds AFAIK.

It may just be my experience but often I've just seen traitors buying pda bombs or silent killing to cripple crew, which is why I say it's often not fun for most people involved though people may disagree on that.

I often see people say they like secret mode where it's one gamemode only, so I'm just testing the waters with this. I personally like dynamic but I do agree it could use some improvement.

It's a draft, I'll fix the CL if I think this is worth pursuing.

**Comment by pedr69** [comment]
the solution is to make traitor items that interact with the blob
emaggable liberators? blob communication headset?

**Comment by BomberBro** [comment]
Solution is to not have any shitass antagonists spawn during blob. Either you're shooting fish in a barrel traitoring or you give it up to help the cripple crew figure out how to rotate an emitter. Blob was not designed with multiple antagonists in mind, a blob headset is an idea but.... I mean, it'd just make the rounds suck even more with traitors explicitly blowing shit up and dismantling the engines to help their blob buddy because blob is a gamemode where the entire crew has to come together. 
>Complicates things? Sure. Arming everyone when someone might not have good intentions IS a complication you should keep in mind. 

"No!"
It's not a gamemode built for that, we already have a ton of gamemodes where you can be a chode and ruin the war effort for the crew side if you're a traitor (revs, cult, xenos, etc) and these at least have the bonus of being able to RP with the other side. Blob is not an RP gamemode, it's PvE and it shouldn't be anything else (nothing wrong with PvE in moderation). 
>B-but then people will know it's extended!

That'll suck for the 1-2 games per year that last 20 minutes past the blob being defeated. Past blob being a virtual guaranteed shuttle call, the PR apparently keeps antags spawning once it's dead.

**Comment by Eneocho** [comment]
> the solution is to make traitor items that interact with the blob emaggable liberators? blob communication headset?

Blob cerebrate spores purchaseable only if tator during a blob round, makes you a cerebrate slaved to the overmind
Emagged liberators reverse the beam polarity, doing next to no damage to blob but extra damage to humanoids.
Psy Relay that lets you communicate directly with the blobs to coordinate shit.
Blob Mindlink spores that let the blob see what you see and perhaps place spores near you.
Blob Psyshield that slaves you to the overmind but lets you walk through the blob walls unimpeded.

File: code/datums/gamemode/dynamic/dynamic.dm [hunk 1]
```diff
@@ -392,6 +392,11 @@ var/stacking_limit = 90
 		// 3. Picking up the CHOSEN ONE.
 		var/datum/dynamic_ruleset/chosen_one = picking_roundstart_rule(drafted_rules, candidate_rules)
 
+		// 4. If Ruleset is exclusive and population is under high_pop_limit, ensure ruleset is only one drafted.
+		if(chosen_one.exclusive && high_pop_limit > rst_pop)
```
Comment (ShiftyRail) [inline code/datums/gamemode/dynamic/dynamic.dm:396]: Could make this a ruleset flag but you don't really need to

File: code/datums/gamemode/dynamic/dynamic.dm [hunk 2]
```diff
@@ -392,6 +392,11 @@ var/stacking_limit = 90
...
+			drafted_rules = list(chosen_one)	
+			break
```
Comment (ShiftyRail) [inline code/datums/gamemode/dynamic/dynamic.dm:398]: I don't like that it skips debug messages

**Comment by tuypo1** [comment]
So if the blob gets killed can things still spawn after that? because if a swiftly killed blob puts it into extended that would be pretty lame

oh wait i'm an idiot you said right there it can add midround stuff

**Comment by DorainGrey** [comment]
I put this out as a feeler, I can see it's not particularly wanted and I do see good points on either side.
I'll probably return to this in the future with more solid ideas.

**Pull request closed without merging**